java/code/src/com/suse/manager/webui/controllers/AnsibleController.java

```diff
@@ -332,7 +334,32 @@ public static String introspectInventory(Request req, Response res, User user) {
 
         try {
             return AnsibleManager.introspectInventory(pathId, user)
-                    .map(inventory -> json(res, success(YAML.dump(inventory))))
+                    .map(inventory -> {
+                        Map<String, Object> data = new HashMap<>();
+
+                        Set<String> hostvars = parseInventoryAndGetHostnames(inventory);
+                        List<SimpleMinionJson> registeredServers = new LinkedList<>();
+                        List<String> unknownHostNames = new LinkedList<>();
+
+                        // fork content into registered or unknown hostnames
+                        hostvars.forEach(
+                                serverName -> {
+                                    if (ServerFactory.findByFqdn(serverName).isPresent()) {
+                                        Server server = ServerFactory.findByFqdn(serverName).get();
+                                        registeredServers.add(new SimpleMinionJson(server.getId(), server.getName()));
+                                    }
+                                    else {
+                                        unknownHostNames.add(serverName);
+                                    }
+                                }
+                        );
```

Comment:
This could be simplified by using `ifPresentOrElse`. As a bonus you avoid
- multiple calls to expensive `findByFqdn`
- `Optional.get()` call.

```suggestion
                        hostvars.forEach(serverName -> ServerFactory.findByFqdn(serverName).ifPresentOrElse(
                                s -> registeredServers.add(new SimpleMinionJson(s.getId(), s.getName())),
                                () -> unknownHostNames.add(serverName))
                        );
```

Reply:
It was exactly my intention but I always forget how to use it and the fact that `ifPresentOrElse` exists :sweat_smile: 
Thank you very much for the pointer :+1: